hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1271,85 +1290,90 @@ bool MDMParser::powerOff(void)
     bool ok = false;
     bool continue_cancel = false;
     bool check_ri = false;
-    if (_init && _pwr) {
-        MDM_INFO("%s = = = = = = = = = = = = = =", POWER_OFF_MSG);
-        if (_cancel_all_operations) {
-            continue_cancel = true;
-            resume(); // make sure we can use the AT parser
+
+    MDM_INFO("%s = = = = = = = = = = = = = =", POWER_OFF_MSG);
+
+    if (!powerState() && !_init && !_pwr) {
+        MDM_INFO("%s Modem is off already", POWER_OFF_MSG);
+        return true;
+    }
+
+    // Make sure we can use the AT parser.
+    // For SARA-G35 it doesn't support powering off the modem using the PWR_ON pin.
+    // But it shares the same code base with other modem, thus, we use AT+CPWROFF generally.
+    if (!(_init && _pwr)) {
+        MDM_INFO("%s Powering on to use AT parser...", POWER_OFF_MSG);
```

Comment:
We are not really powering on here, we are trying to do some low level initialization to talk to the modem.